plugins/inputs/ctrlx_datalayer/ctrlx_datalayer_test.go

```diff
@@ -101,13 +100,10 @@ func TestCtrlXCreateSubscriptionDriven(t *testing.T) {
 				Log:          testutil.Logger{},
 			}
 			subID, err := s.createSubscription(&subs[0])
-
-			if test.subID != "" {
-				require.NoError(t, err)
-				require.NotEmpty(t, subID)
-			} else {
-				require.Error(t, err)
+			if err != nil {
 				require.Empty(t, subID)
+			} else {
+				require.NotEmpty(t, subID)
 			}
```

Comment:
This is a bit weird, I don't know if this is a good test. As currently you could have an error with the status 200 test, but that would be fine as long as the returned subID would be empty?

I think you should have a boolean in the testcase to indicate there should be an error or not. Even better would be a string that should be contained in the error. There are special `require` methods that can work with that.

Reply:
We modified the test, see https://github.com/influxdata/telegraf/pull/11155/commits/a47c7896129cc8e5f7459d84639ce1d6a0842db6